TAO/orbsvcs/ImplRepo_Service/AsyncAccessManager.cpp

```diff
@@ -283,11 +283,27 @@ AsyncAccessManager::update_status (ImplementationRepository::AAM_Status s)
 }
 
 void
-AsyncAccessManager::activator_replied (bool success)
+AsyncAccessManager::activator_replied (bool success, pid_t pid)
 {
   if (success)
     {
-      this->update_status (ImplementationRepository::AAM_WAIT_FOR_RUNNING);
+      if (pid == 0)
```

Comment:
While pid 0 may be a kernel pager or scheduler process, using 0 to mean "no pid" is a common thing. Consider posix fork() where the function returns 0 in the child process or -1 (invalid pid) on an error.